### ISSUE TITLE
non-fixed "fx" files - causes discontinuity in timeseries between experiments

### ISSUE THREAD

**Issue by ChrisJones-MOHC** [ESMValGroup/ESMValCore#1474]
I have discovered that not all “fx” files are constant for a given model on ESGF. Specific example is for sftlf (land fraction) for NorESM2-LM model data below. This leads to discontinuity in global timeseries if sftlf is used to weight variables such as land carbon stores or fluxes

This is not a ESMVal error, but a danger with data which is useful for users to be aware of

Example – I have not examined all occurrences of sftlf but browsed a few for models where I have spotted an issue. The following for example show that NorESM2-LM has at least 3 different sftlf files. Pasted below are the filenames (including version dates) and the global average. (CNRM-ESM2-1/piControl/r1i1p1f2/fx/sftlf/gr/v20181115/sftlf_fx_CNRM-ESM2-1_piControl_r1i1p1f2_gr.nc also appears to differ from other CNRM-ESM2-1 files)

CMIP6/CMIP/NCC/NorESM2-LM/historical/r2i1p1f1/fx/sftlf/gn/v20190920/sftlf_fx_NorESM2-LM_historical_r2i1p1f1_gn.nc	28.340188318499596
CMIP6/CMIP/NCC/NorESM2-LM/historical/r3i1p1f1/fx/sftlf/gn/v20190920/sftlf_fx_NorESM2-LM_historical_r3i1p1f1_gn.nc	28.340188318499596
CMIP6/CMIP/NCC/NorESM2-LM/piControl/r1i1p1f1/fx/sftlf/gn/v20210118/sftlf_fx_NorESM2-LM_piControl_r1i1p1f1_gn.nc		28.340188318499596
CMIP6/CMIP/NCC/NorESM2-LM/historical/r2i1p1f1/fx/sftlf/gn/v20190920/sftlf_fx_NorESM2-LM_historical_r2i1p1f1_gn.nc	28.340188318499596
CMIP6/CMIP/NCC/NorESM2-LM/historical/r3i1p1f1/fx/sftlf/gn/v20190920/sftlf_fx_NorESM2-LM_historical_r3i1p1f1_gn.nc	28.340188318499596
CMIP6/CMIP/NCC/NorESM2-LM/piControl/r1i1p1f1/fx/sftlf/gn/v20210118/sftlf_fx_NorESM2-LM_piControl_r1i1p1f1_gn.nc		28.340188318499596

CMIP6/CMIP/NCC/NorESM2-LM/1pctCO2/r1i1p1f1/fx/sftlf/gn/v20190815/sftlf_fx_NorESM2-LM_1pctCO2_r1i1p1f1_gn.nc		28.34260542007538
CMIP6/CMIP/NCC/NorESM2-LM/historical/r1i1p1f1/fx/sftlf/gn/v20190815/sftlf_fx_NorESM2-LM_historical_r1i1p1f1_gn.nc	28.34260542007538
CMIP6/CMIP/NCC/NorESM2-LM/1pctCO2/r1i1p1f1/fx/sftlf/gn/v20190815/sftlf_fx_NorESM2-LM_1pctCO2_r1i1p1f1_gn.nc		28.34260542007538

CMIP6/ScenarioMIP/NCC/NorESM2-LM/ssp126/r1i1p1f1/fx/sftlf/gn/v20191108/sftlf_fx_NorESM2-LM_ssp126_r1i1p1f1_gn.nc	29.352518609708902
CMIP6/ScenarioMIP/NCC/NorESM2-LM/ssp370/r1i1p1f1/fx/sftlf/gn/v20191108/sftlf_fx_NorESM2-LM_ssp370_r1i1p1f1_gn.nc	29.352518609708902
CMIP6/ScenarioMIP/NCC/NorESM2-LM/ssp370/r2i1p1f1/fx/sftlf/gn/v20191108/sftlf_fx_NorESM2-LM_ssp370_r2i1p1f1_gn.nc	29.352518609708902
CMIP6/ScenarioMIP/NCC/NorESM2-LM/ssp370/r3i1p1f1/fx/sftlf/gn/v20191108/sftlf_fx_NorESM2-LM_ssp370_r3i1p1f1_gn.nc	29.352518609708902
CMIP6/ScenarioMIP/NCC/NorESM2-LM/ssp126/r1i1p1f1/fx/sftlf/gn/v20191108/sftlf_fx_NorESM2-LM_ssp126_r1i1p1f1_gn.nc	29.352518609708902
CMIP6/ScenarioMIP/NCC/NorESM2-LM/ssp370/r1i1p1f1/fx/sftlf/gn/v20191108/sftlf_fx_NorESM2-LM_ssp370_r1i1p1f1_gn.nc	29.352518609708902
CMIP6/ScenarioMIP/NCC/NorESM2-LM/ssp370/r2i1p1f1/fx/sftlf/gn/v20191108/sftlf_fx_NorESM2-LM_ssp370_r2i1p1f1_gn.nc	29.352518609708902
CMIP6/ScenarioMIP/NCC/NorESM2-LM/ssp370/r3i1p1f1/fx/sftlf/gn/v20191108/sftlf_fx_NorESM2-LM_ssp370_r3i1p1f1_gn.nc	29.352518609708902

**Comment by zklaus** [comment]
Thank. Please also note the following erratum: https://errata.es-doc.org/static/view.html?uid=0b9d744d-be7c-0152-89fc-ca64ab2dbe40

To quote:
> all the sftlf and sftgif data that have version numbers v20191108 and earlier should use those from the abrupt-4xCO2 experiment of the LM and MM versions respectively.

**Comment by ChrisJones-MOHC** [comment]
thanks Klaus - useful to know this has been flagged there.

Although it's easier said than done, because unless you know this the tool defaults to using the experiment you've selected. Is it possible to redact bad files rather than leave them in place and hope everyone reads the errata?

**Comment by zklaus** [comment]
Yes, that is possible. However, it must be done by the modeling groups.

**Comment by ChrisJones-MOHC** [comment]
understood! I have emailed the contact email in the netcdf files, noresm-ncc@met.no

**Comment by zklaus** [comment]
Cheers, @ChrisJones-MOHC. Let us know how it goes.

**Comment by ChrisJones-MOHC** [comment]
update - NorESM-LM will retract incorrect sftlf files. https://github.com/NorESMhub/noresm2cmor/issues/162 refers

in addition - use of the following line in ESMVal pre-processor forces the tool to use a correct (in this case abrupt-4xCO2) file:
fx_variables: [{short_name: sftlf, mip: fx, exp: abrupt-4xCO2}]

**Comment by YanchunHe** [comment]
Hi, just want to mention, seems that the abrupt-4xCO2 by NorESM2-LM is not correct neither, as the whole dataset has been retracted, merged and republished. By anyhow, the republished v20210118  `sftlf` of `abrupt-4xCO2` used a wrong version.

As mentioned in the https://github.com/NorESMhub/noresm2cmor/issues/162 issue, the v20210118 version will be retracted from ESGF, including the abrupt-4xCO2.

A correct version for NorESM2-LM piControl, historical and amip will be published later. And all other dataset that will not be retracted will be correct, hopefully.

**Comment by rswamina** [comment]
@bouweandela and @zklaus - What is the best way to record the outcome from this discussion on what files to use? Should users facing this problem search the issues or should we be documenting this somewhere?

**Comment by YanchunHe** [comment]
Now the wrong version of `sftlf` dataset has been retracted, and republished with correct version of `v20191108`, for `NorESM2-LM`, experiments, `piControl`, `historical` and `amip`;

for ESMVal pr-processor, one can use, e.g., 
fx_variables: [{short_name: sftlf, mip: fx, exp: piControl}]

We may close this issue now. Thanks for the patience!
